plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts

```diff
@@ -190,14 +225,20 @@ export function createFetchTemplateAction(options: {
       );
 
       for (const location of allEntriesInTemplate) {
-        const shouldCopyWithoutRender = nonTemplatedEntries.has(location);
+        let shouldCopyWithoutRender = nonTemplatedEntries.has(location);
 
-        const outputPath = resolvePath(
-          outputDir,
-          shouldCopyWithoutRender
-            ? location
-            : templater.renderString(location, context),
-        );
+        let localOutputPath = location;
+        if (extension) {
+          if (extname(localOutputPath) === extension) {
+            localOutputPath = localOutputPath.slice(0, -extension.length);
+          } else {
+            shouldCopyWithoutRender = true;
+          }
+          localOutputPath = templater.renderString(localOutputPath, context);
+        } else if (!shouldCopyWithoutRender) {
+          localOutputPath = templater.renderString(localOutputPath, context);
+        }
+        const outputPath = resolvePath(outputDir, localOutputPath);
 
         if (shouldCopyWithoutRender) {
```

Comment:
nit: the log statement here is not quite accurate any more, since it could be that rendering is being skipped because of the file extension, rather than `copyWithoutRender`.